src/main/java/com/ubc/cpsc304/hotelify/controller/AdministratorController.java

```diff
@@ -0,0 +1,101 @@
+package com.ubc.cpsc304.hotelify.controller;
+
+import com.ubc.cpsc304.hotelify.controller.dto.AdministratorRequestDto;
+import com.ubc.cpsc304.hotelify.controller.dto.AdministratorResponseDto;
+import com.ubc.cpsc304.hotelify.controller.dto.AuthenticationRequestDto;
+import com.ubc.cpsc304.hotelify.controller.dto.AuthenticationResponseDto;
+import com.ubc.cpsc304.hotelify.entity.Administrator;
+import com.ubc.cpsc304.hotelify.exception.ConflictException;
+import com.ubc.cpsc304.hotelify.exception.NotFoundException;
+import com.ubc.cpsc304.hotelify.exception.UnauthorizedException;
+import com.ubc.cpsc304.hotelify.service.AdministratorService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Administrator Controller
+ * Created by Rex on 2018-11-05
+ */
+@Slf4j
+@RestController
+@RequestMapping("/administrators")
+public class AdministratorController {
+
+    private static final String ADMINISTRATOR_TYPE = "admin";
+
+    private final AdministratorService administratorService;
+
+    public AdministratorController(AdministratorService administratorService) {
+        this.administratorService = administratorService;
+    }
+
+    @GetMapping(path = "/{username}")
+    private AdministratorResponseDto getAdministrator(@PathVariable String username)
+            throws NotFoundException {
+
+        Administrator administrator = this.administratorService.getAdministrator(username);
+
+        return AdministratorController.convertModel(administrator);
+    }
+
+    @PostMapping("/login")
+    private AuthenticationResponseDto login(
+            @RequestBody AuthenticationRequestDto authenticationRequestDto
+    ) throws NotFoundException, UnauthorizedException {
+
+        this.administratorService.login(authenticationRequestDto);
+
+        AuthenticationResponseDto authenticationResponseDto = new AuthenticationResponseDto();
+
+        authenticationResponseDto.setUserType(ADMINISTRATOR_TYPE);
+
+        return authenticationResponseDto;
+    }
+
+    @PostMapping
+    private AdministratorResponseDto createAdministrator(
+            @RequestBody AdministratorRequestDto administratorRequestDto
+    ) throws ConflictException {
+
+        Administrator administrator = this.administratorService.createAdministrator(
+                administratorRequestDto
+        );
+
+        return AdministratorController.convertModel(administrator);
+    }
+
+    @PutMapping
```

Comment:
is it necessary since we can get it from the request body?

Reply:
in short, In this way it follows the RESTful API naming convention

Reply:
`user` is kind of an exception case. Imagine instead of `username` as the identifier, you use some random id as the identifier, which should not exist in your `requestDto`. This is kind of the reason why you should specify the `username` in the url, but not the body,

Reply:
ok